apps/src/templates/teacherDashboard/AssignmentSelector.jsx

```diff
@@ -76,10 +83,17 @@ export default class AssignmentSelector extends Component {
 
   onChangePrimary = (event) => {
     const { assignments, section } = this.props;
-    const currentSecondaryId = assignmentId(null, section.scriptId);
+    let currentSecondaryId;
+    if (!section) {
+      currentSecondaryId = noAssignment;
+    } else {
+      currentSecondaryId = assignmentId(null, section.scriptId);
+    }
```

Comment:
Changing a primary should clear your scriptId I think. i.e. If I change my primary from CSD to CSP, I dont want to still have csd1 assigned.